clippy_lints/src/casts/fn_to_numeric_cast_any.rs

```diff
@@ -8,12 +8,7 @@ use rustc_middle::ty::{self, Ty};
 use super::FN_TO_NUMERIC_CAST_ANY;
 
 pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, cast_expr: &Expr<'_>, cast_from: Ty<'_>, cast_to: Ty<'_>) {
-    // We allow casts from any function type to any function type.
-    match cast_to.kind() {
-        ty::FnDef(..) | ty::FnPtr(..) => return,
-        _ => { /* continue to checks */ },
-    }
-
+    assert!(cast_to.is_numeric());
```

Comment:
Is that a leftover? We tend not to use `assert!()` in Clippy itself. As you moved the call to this function, it should not be called with non-numeric cast targets, or the stderr will change anyway and this will be caught by the CI.